File: ai_service/terminal_log.py
"""Colored terminal logging for AI service startup."""

from __future__ import annotations

_C = {
    "reset": "\033[0m",
    "bold": "\033[1m",
    "dim": "\033[2m",
    "green": "\033[32m",
    "yellow": "\033[33m",
    "red": "\033[31m",
    "cyan": "\033[36m",
    "blue": "\033[34m",
    "magenta": "\033[35m",
}


def banner(title: str) -> None:
    line = "═" * max(20, len(title) + 4)
    print(f"\n{_C['cyan']}{_C['bold']}╔{line}╗{_C['reset']}")
    print(f"{_C['cyan']}{_C['bold']}║  {title:<{len(line)-2}}║{_C['reset']}")
    print(f"{_C['cyan']}{_C['bold']}╚{line}╝{_C['reset']}")


def ok(tag: str, message: str) -> None:
    print(f"{_C['green']}✓{_C['reset']} {_C['bold']}{tag}{_C['reset']} {message}")


def warn(tag: str, message: str) -> None:
    print(f"{_C['yellow']}⚠{_C['reset']} {_C['bold']}{tag}{_C['reset']} {message}")


def err(tag: str, message: str) -> None:
    print(f"{_C['red']}✗{_C['reset']} {_C['bold']}{tag}{_C['reset']} {message}")


def info(tag: str, message: str) -> None:
    print(f"{_C['blue']}●{_C['reset']} {_C['bold']}{tag}{_C['reset']} {message}")


def dim(message: str) -> None:
    print(f"{_C['dim']}  {message}{_C['reset']}")


def voice(tag: str, message: str) -> None:
    print(f"{_C['magenta']}♪{_C['reset']} {_C['bold']}{tag}{_C['reset']} {message}")


def row(label: str, value: str, ok: bool | None = None) -> None:
    if ok is True:
        icon = f"{_C['green']}●{_C['reset']}"
    elif ok is False:
        icon = f"{_C['red']}●{_C['reset']}"
    else:
        icon = f"{_C['dim']}○{_C['reset']}"
    print(f"  {icon} {_C['dim']}{label}:{_C['reset']} {value}")

File: backend/src/utils/terminalLog.ts
/**
 * Colored, consistent terminal output for ELVA backend startup and voice pipeline.
 */

const c = {
    reset: '\x1b[0m',
    bold: '\x1b[1m',
    dim: '\x1b[2m',
    green: '\x1b[32m',
    yellow: '\x1b[33m',
    red: '\x1b[31m',
    cyan: '\x1b[36m',
    blue: '\x1b[34m',
    magenta: '\x1b[35m',
    white: '\x1b[37m',
};

export const terminalLog = {
    banner(title: string): void {
        const line = '═'.repeat(Math.max(20, title.length + 4));
        console.log(`\n${c.cyan}${c.bold}╔${line}╗${c.reset}`);
        console.log(`${c.cyan}${c.bold}║  ${title.padEnd(line.length - 2)}║${c.reset}`);
        console.log(`${c.cyan}${c.bold}╚${line}╝${c.reset}`);
    },

    ok(tag: string, message: string): void {
        console.log(`${c.green}✓${c.reset} ${c.bold}${tag}${c.reset} ${message}`);
    },

    warn(tag: string, message: string): void {
        console.log(`${c.yellow}⚠${c.reset} ${c.bold}${tag}${c.reset} ${message}`);
    },

    err(tag: string, message: string): void {
        console.log(`${c.red}✗${c.reset} ${c.bold}${tag}${c.reset} ${message}`);
    },

    info(tag: string, message: string): void {
        console.log(`${c.blue}●${c.reset} ${c.bold}${tag}${c.reset} ${message}`);
    },

    voice(tag: string, message: string): void {
        console.log(`${c.magenta}♪${c.reset} ${c.bold}${tag}${c.reset} ${message}`);
    },

    dim(message: string): void {
        console.log(`${c.dim}  ${message}${c.reset}`);
    },

    row(label: string, value: string, ok?: boolean): void {
        const icon = ok === true ? `${c.green}●${c.reset}` : ok === false ? `${c.red}●${c.reset}` : `${c.dim}○${c.reset}`;
        console.log(`  ${icon} ${c.dim}${label}:${c.reset} ${value}`);
    },
};
